src/library/client.lua

```diff
@@ -33,81 +113,46 @@ function client.get_raw_finale_version(major, minor, build)
 end
 
 --[[
-% supports_smufl_fonts()
+% supports
 
-Returns true if the current client supports SMuFL fonts.
-
-: (boolean)
-]]
-function client.supports_smufl_fonts()
-    return finenv.RawFinaleVersion >= client.get_raw_finale_version(27, 1)
-end
+Checks the client supports a given feature. Returns true if the client
+supports the feature, false otherwise.
 
---[[
-% supports_category_save_with_new_type()
+To assert the client must support a feature, use `client.assert_supports`.
 
-Returns true if the current client supports FCCategory::SaveWithNewType().
+For a list of valid features, see the [`features` table in the codebase](https://github.com/finale-lua/lua-scripts/blob/master/src/library/client.lua#L52).
 
+@ feature (string) The feature the client should support.
 : (boolean)
 ]]
-function client.supports_category_save_with_new_type()
-    return finenv.StringVersion >= "0.58"
+function client.supports(feature)
+    return features[feature].test()
 end
 
 --[[
-% supports_finenv_query_invoked_modifier_keys()
+% assert_supports
 
-Returns true if the current client supports finenv.QueryInvokedModifierKeys().
+Asserts that the client supports a given feature. If the client doesn't
+support the feature, this function will throw an friendly error then
+exit the program.
 
-: (boolean)
-]]
-function client.supports_finenv_query_invoked_modifier_keys()
-    return finenv.IsRGPLua and finenv.QueryInvokedModifierKeys
-end
+To simply check if a client supports a feature, use `client.supports`.
 
---[[
-% supports_retained_state()
-
-Returns true if the current client supports retaining state between runs.
-
-: (boolean)
-]]
-function client.supports_retained_state()
-    return finenv.IsRGPLua and finenv.RetainLuaState ~= nil
-end
-
---[[
-% supports_modeless_dialog()
-
-Returns true if the current client supports modeless dialogs.
+For a list of valid features, see the [`features` table in the codebase](https://github.com/finale-lua/lua-scripts/blob/master/src/library/client.lua#L52).
 
+@ feature (string) The feature the client should support.
 : (boolean)
 ]]
-function client.supports_modeless_dialog()
-    return finenv.IsRGPLua
-end
-
---[[
-% supports_clef_changes()
-
-Returns true if the current client supports changing clefs.
-
-: (boolean)
-]]
-function client.supports_clef_changes()
-    return finenv.IsRGPLua or finenv.StringVersion >= "0.60"
-end
-
---[[
-% supports_custom_key_signatures()
-
-Returns true if the current client supports changing clefs.
-
-: (boolean)
-]]
-function client.supports_custom_key_signatures()
-    local key = finale.FCKeySignature()
-    return finenv.IsRGPLua and key.CalcTotalChromaticSteps
+function client.assert_supports(feature)
+    local error_code = finenv.DebugEnabled and 2 or 0
```

Comment:
Apologies for the nit, but for correctness this should be named `error_level`

Reply:
True. In other languages it's an error "code", but it does look like it's "level" in Lua.

https://www.lua.org/manual/5.1/manual.html#pdf-error

Reply:
And no worries about the small nits. Things like this matter in the long run.